packages/addon-dev/src/rollup-hbs-plugin.ts

```diff
@@ -4,19 +4,48 @@ import { readFileSync } from 'fs';
 const backtick = '`';
 
 export default function rollupHbsPlugin(): Plugin {
-  const filter = createFilter('**/*.hbs');
+  const templateFilter = createFilter('**/*.hbs');
+  const moduleFilter = createFilter('**/*.hbs.js');
 
   return {
     name: 'rollup-hbs-plugin',
+    async resolveId(source: string, importer: string | undefined, options) {
+      const resolution = await this.resolve(source, importer, {
+        skipSelf: true,
+        ...options,
+      });
+
+      const id = resolution?.id;
+
+      if (!templateFilter(id)) return null;
+
+      // This creates an `*.hbs.js` that we will populate in `load()` hook.
+      return {
+        ...resolution,
+        id: id + '.js',
+        meta: {
+          'rollup-hbs-plugin': {
+            originalId: id,
+          },
+        },
+      };
+    },
     load(id: string) {
-      if (!filter(id)) return;
-      let input = readFileSync(id, 'utf8');
+      if (!moduleFilter(id)) return null;
+
+      const meta = this.getModuleInfo(id)?.meta;
+      const originalId = meta?.['rollup-hbs-plugin']?.originalId;
+
+      if (!originalId) {
+        return;
+      }
```

Comment:
I think we can remove the `moduleFilter` conditional, because the thing we _actually_ care about is that the meta contains the original id path info (and we return early anyways).

```suggestion
      const meta = this.getModuleInfo(id)?.meta;
      const originalId = meta?.['rollup-hbs-plugin']?.originalId;

      if (!originalId) {
        return;
      }
```

Reply:
Updated my PR.